onnx/defs/schema.cc

```diff
@@ -735,26 +737,31 @@ void OpSchema::UpdateFunctionProtoOpsetImportVersion(FunctionProto& function_pro
 bool OpSchema::BuildContextDependentFunctionWithOpsetVersion(
     const FunctionBodyBuildContext& ctx,
     FunctionProto& function_proto,
-    int opset_version) const {
+    int opset_version,
+    const std::string& domain,
+    bool fail_on_invalid_op) const {
   if (opset_version == OpSchema::kUninitializedSinceVersion)
     opset_version = since_version_;
 
   std::map<int, ContextDependentFunctionBodyBuilder>::const_iterator it =
-      opset_version_to_function_builder_.find(opset_version);
-
-  if (it != opset_version_to_function_builder_.end()) {
+      opset_version_to_function_builder_.upper_bound(opset_version);
+  if (opset_version_to_function_builder_.empty() || it == opset_version_to_function_builder_.begin()) {
+    ONNX_THROW_EX(std::out_of_range(
```

Comment:
Let's avoid the exception. Return false if unable to build a function. If we want to return extra information, we should return an extra error message.